esphome/components/mcp4461/mcp4461.cpp

```diff
@@ -0,0 +1,638 @@
+  reg |= static_cast<uint8_t>(Mcp4461EepromLocation::MCP4461_EEPROM_1) + (static_cast<uint8_t>(location) * 0x10);
+  reg |= static_cast<uint8_t>(Mcp4461Commands::READ);
+  uint16_t buf;
+  if (!this->is_eeprom_ready_for_writing_(true)) {
+    return 0;
+  }
+  if (!this->read_byte_16(reg, &buf)) {
+    this->error_code_ = MCP4461_STATUS_I2C_ERROR;
+    this->status_set_warning();
+    ESP_LOGW(TAG, "Error fetching EEPRom location value");
+    return 0;
+  }
+  return buf;
+}
+
+bool Mcp4461Component::set_eeprom_value(Mcp4461EepromLocation location, uint16_t value) {
+  if (this->is_failed()) {
+    ESP_LOGE(TAG, "%s", LOG_STR_ARG(this->get_message_string(this->error_code_)));
+    return false;
+  }
+  uint8_t addr = 0;
+  if (value > 511) {
+    return false;
+  }
+  if (value > 256) {
+    addr = 1;
+  }
+  addr |= static_cast<uint8_t>(Mcp4461EepromLocation::MCP4461_EEPROM_1) + (static_cast<uint8_t>(location) * 0x10);
+  if (!(this->mcp4461_write_(addr, value, true))) {
+    this->error_code_ = MCP4461_STATUS_I2C_ERROR;
+    this->status_set_warning();
+    ESP_LOGW(TAG, "Error writing EEPROM value");
+    return false;
+  }
+  return true;
+}
+
+bool Mcp4461Component::is_writing_() {
+  /* Read the EEPROM write-active status from the status register */
+  bool writing = static_cast<bool>((this->get_status_register_() >> 4) & 0x01);
+
+  /* If EEPROM is no longer writing, reset the timeout flag */
+  if (!writing) {
+    /* This is protected boolean flag in Mcp4461Component class */
+    this->last_eeprom_write_timed_out_ = false;
+  }
+
+  return writing;
+}
+
+bool Mcp4461Component::is_eeprom_ready_for_writing_(bool wait_if_not_ready) {
+  /* Check initial write status */
+  bool ready_for_write = !this->is_writing_();
+
+  /* Return early if no waiting is required or EEPROM is already ready */
+  if (ready_for_write || !wait_if_not_ready || this->last_eeprom_write_timed_out_) {
+    return ready_for_write;
+  }
+
+  /* Timestamp before starting the loop */
+  const uint32_t start_millis = millis();
+
+  ESP_LOGV(TAG, "Waiting until EEPROM is ready for write, start_millis = %" PRIu32, start_millis);
+
+  /* Loop until EEPROM is ready or timeout is reached */
+  while (!ready_for_write && ((millis() - start_millis) < EEPROM_WRITE_TIMEOUT_MS)) {
+    ready_for_write = !this->is_writing_();
+
+    /* If ready, exit early */
+    if (ready_for_write) {
+      ESP_LOGV(TAG, "EEPROM is ready for new write, elapsed_millis = %" PRIu32, millis() - start_millis);
+      return true;
+    }
+
+    /* Not ready yet, yield before checking again */
+    yield();
+  }
+
+  /* If still not ready after timeout, log error and mark the timeout */
+  ESP_LOGE(TAG, "EEPROM write timeout exceeded (%u ms)", EEPROM_WRITE_TIMEOUT_MS);
+  this->last_eeprom_write_timed_out_ = true;
+
+  return false;
+}
+
+bool Mcp4461Component::mcp4461_write_(uint8_t addr, uint16_t data, bool nonvolatile) {
+  uint8_t reg = 0;
+  if (data > 0xFF) {
+    reg = 1;
+  }
```

Comment:
Found one more of these, haha:
```suggestion
  uint8_t reg = data > 0xff ? 1 : 0;
```